### PR TITLE
Allow more granular composition of rule configuration

### PR DESCRIPTION
**Pull request by nchen63**
#### PR checklist

- [x] Addresses an existing issue: #2278
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Overview of change:

Can override just severity or just options
Refactor to parse/upgrade/standardize config up front

#### CHANGELOG.md entry:

[no-log] this follows #1738

### CODE REVIEW

File: src/language/rule/abstractRule.ts [hunk 1]
```diff
@@ -17,38 +17,24 @@
...
-        } else {
-            this.ruleArguments = [];
-        }
-
-        if (value.options) {
-            this.ruleArguments = arrayify(value.options);
-        }
-
-        this.ruleSeverity = getRuleSeverity(value);
+    constructor(private options: IOptions) {
```
Comment (adidahiya) [inline src/language/rule/abstractRule.ts]: this is another API break for 5.0 too, right? let's make sure to document the upgrade steps for custom rules

**Comment by adidahiya** [comment]
should we update the tslint:recommended and tslint:latest to use the new config syntax?

File: src/language/rule/abstractRule.ts (hunk 1 above)
Comment (nchen63) [inline src/language/rule/abstractRule.ts]: I don't think this should affect custom rules. Nobody should be calling the constructor except for ruleLoader in tslint itself.

Comment (ajafff) [inline src/language/rule/abstractRule.ts]: But you can still override the constructor in a base class

**Pull request merged**